File: backend/chess4connect/chess4connect/Models/Database/Entities/Play.cs
using chess4connect.Enums;
using chess4connect.Models.Games;
using System.ComponentModel.DataAnnotations.Schema;

namespace chess4connect.Models.Database.Entities;

public class Play
{
    public int Id { get; set; }
    public int WinnerId { get; set; }
    public GameResult GameResult { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public GameType Game { get; set; }
    public List<PlayDetail> PlayDetails { get; set; } = new List<PlayDetail>();
}

File: frontend/chessandconnect/src/app/models/play.ts
import { playState } from "../enums/playState";
import { User } from "./dto/user";
import { GameType } from "../enums/game";

export interface Play {
    id: number,
    user: User,
    opponent: User,
    duration: number
    playState: playState,
    game: GameType
}
